ome/omero-iviewer
# Thumbnail scrolling

See https://trello.com/c/JLpfTbmW/4-improve-thumbnails-loading-display

Instead of clicking up/down to load more thumbnails, the thumbnail panel is created at the full size (with only the visible thumbnails loaded) and more thumbnails are loaded when we scroll.
There are lots of different cases to handle and a lot has changed so everything needs testing:

 - Expected behaviour: any thumbnails that appear in the panel should be loaded (no empty grey squares).
 - While actually scrolling we don't load thumbnails, since that would load the whole dataset if you scroll fast from one end to the other. Instead, wait until scrolling slows / stops, then load thumbs for visible thumbnails.
 - You can see when loading is happening in the devtools Network tab.
 - In all scenarios below, check thumbs on load, then scroll about, then refresh thumbs (when scrollbar is in various locations).
 - Check that IE works.

**Scenarios to test:**

 - Single image in 1 or more Datasets, double-clicked in webclient. url like ```/webclient/img_detail/4413/?dataset=1301```
        - Check that thumbs loaded from *current* parent Dataset (not a different Dataset that the image is in)
        - Try double-clicking images in webclient near the start of the Dataset and at the end. Thumb panel in iviewer should scroll to the show the selected image.
        - Try with a small Dataset (< 5 images) and as large as possible.
        - Remove the ```?datasets=ID``` from the URL and reload. If image is in single Dataset, thumbs should be loaded. If more than 1 Dataset, thumbnail panel hidden.
 - Multiple selected images > Open with iviewer. Url like ```/iviewer/?images=4402,4403,4404,4405 ```
 - Orphaned image, double-clicked in webclient. e.g. ```webclient/img_detail/57012/``` should hide thumbnail panel.
 - Well - openwith > iviewer.  Url ```/iviewer/?well=148``` First image should be selected.
 - Datasets - openwith > iviewer. ```/iviewer/?dataset=10480``` First image selected.
